src/cmap/auth/kerberosAuthProcess.ts

```diff
@@ -0,0 +1,185 @@
+const dns = require('dns');
+import { Kerberos } from '../../deps';
+
+type TransitionCallback = (err?: Error | null, payload?: any) => void;
+type Transition = typeof MongoAuthProcess.prototype.transition;
+
+interface KerberosClient {
+  step: (challenge: string, callback?: TransitionCallback) => Promise<string> | void;
+  wrap: (
+    challenge: string,
+    options?: { user: string },
+    callback?: TransitionCallback
+  ) => Promise<string> | void;
+  unwrap: (challenge: string, callback?: TransitionCallback) => Promise<string> | void;
+}
+
+interface gssapiOptions {
+  gssapiServiceName?: string;
+  gssapiCanonicalizeHostName?: boolean;
+}
+
+export class MongoAuthProcess {
```

Comment:
I was hoping there was a way to merge this directly with the `gssapi.ts` code, there's no good reason to abstract it anymore.